ledger/complete/mtrie/trie/trie.go

```diff
@@ -104,18 +104,18 @@ func (mt *MTrie) String() string {
 // UnsafeRead read payloads for the given paths. It is called unsafe as it requires the
 // paths to be sorted
 // TODO move consistency checks from Forrest into Trie to obtain a safe, self-contained API
-func (mt *MTrie) UnsafeRead(paths []ledger.Path) ([]*ledger.Payload, error) {
```

Comment:
not sure why we remove the error like here, ideally, we should keep the error line here for unhappy cases and potential attacks to be handled gracefully by exec node than panicing.

Reply:
Yes I removed it because the function never errors. Keeping the error in low level functions results in many unnecessary comparisons and branching (the if statements). 
I totally agree that if there is an error scenario, we should return the error and not panic.